src/RandUtil.chpl

```diff
@@ -0,0 +1,90 @@
+module RandUtil {
+    use MultiTypeSymbolTable;
+    use MultiTypeSymEntry;
+    use RandMsg;
+    use ArkoudaRandomCompat;
+
+    const minPerStream = 256; // minimum number of elements per random stream
+
+    record scalarOrArray {
+        var isArray: bool;
+        var sym;  // TODO figure out type hint here to avoid generic
+        var val: real;
+
+        proc init(scalarOrArrayString: string, isArray: bool, st: borrowed SymTab) {
+            // I'm not sure if there's a good way to remove these try!
```

Comment:
Throwing initializers are currently somewhat incomplete as a feature, but you can call a throwing helper procedure from within an initializer after `init this`. Ex:

```chapel
proc init() throws {
  this.x = 1;
  init this;
  this.y = throwingHelper();
}
```

Which isn't the cleanest pattern, but could allow you to remove the `try!`'s here.

Reply:
that's what I thought about throwing initializers. And I like this better than using `try!` so I'll update that!

Reply:
Hmmm I tried this and it wasn't very happy

```chapel
/proj/arkouda//src/RandUtil.chpl:14: In initializer:
/proj/arkouda//src/RandUtil.chpl:19: error: call to throwing function throwingSymEntryHelper without throws, try, or try! (relaxed mode)
/proj/arkouda//src/RandUtil.chpl:33: note: throwing function throwingSymEntryHelper defined here
```

Reply:
ooops I missed that ur `init` had a `throws` after it! retyring now

Reply:
I couldn't figure out the type hint that `sym` should have and it was angry that `sym` hadn't be initialized since it has no type
```
proj/arkouda//src/RandUtil.chpl:14: In initializer:
proj/arkouda//src/RandUtil.chpl:17: error: can't omit initialization of field "sym", no type or default value provided
```

I tried defualting it to `makeDistArray([0.0])` but that caused this error with runtime checks enabled:
`error: halt reached - assigning between arrays of different shapes in dimension 0: 1 vs. 5`

so it seems like the only one I could easily add a throwing helper to is the cast to real. I think I'm just gonna leave this as is for now